src/lib/support/CHIPMemString.h

```diff
@@ -84,9 +84,16 @@ inline void CopyString(char * dest, size_t destLength, ByteSpan source)
 {
     if (dest && destLength)
```

Comment:
nit, can close if a hassle (since the if is already existing): I find this style of nested ifs more difficult to reason through. I think early checks and bail outs may be easier on readers, like:

```cpp
if ((dest == nullptr) || (destLength == 0))
{
    return; // no space to copy anything, not even a null terminator
}

if (source.empty()) 
{
    *dest = 0;  // just a null terminator, we are copying empty data
    return;
}

size_t maxChars = std::min(destLength - 1, source.size());
memcpy(dest, source.data(), maxChars);
dest[maxChars] = '\0';
```

Totally obviously subjective here. Your code is correct.

Reply:
I agree that it looks better, integrated.